spec/rspec/core_spec.rb

```diff
@@ -13,6 +13,9 @@
   # JRuby appears to not respect `--disable=gem` so rubygems also gets loaded.
   allowed_loaded_features << /rubygems/ if RSpec::Support::Ruby.jruby?
 
+  # Truffleruby cext files
+  allowed_loaded_features << /truffle\/cext/ if RSpec::Support::Ruby.truffleruby?
```

Comment:
This makes sense (the first C ext required loads a couple Ruby files from stdlib), it could be `/\/truffle\/cext/` to be a bit more precise.